### DISCUSSION TITLE
Is it possible to only use sketch (editor) without engine?

### DISCUSSION THREAD

**Discussion by linonetwo** [Blackprint/Blackprint#12] [Q&A] [answered]
Hi, I want to create a flow based visual editor for tiddlywiki's action widget and [filter expression](https://github.com/Jermolene/TiddlyWiki5/issues/5058).

The execution of the flow will be handled by tiddlywiki, so I want to only bring the editor to tiddlywiki for minimal size. Is this possible? And do you have idea how much size will it be? Thanks!

**Comment by StefansArya** [comment, marked as answer]
Hi!
Sorry, but it's not possible for now as the sketch depends on the engine.
The cable connection and data synchronization is handled by the engine, and the sketch is the editor of the engine.

https://github.com/Blackprint/Blackprint/blob/8d591f0fc5460eae1075548e36d35521a187a7f0/src/_init.js#L12-L15

About the sketch size including the engine + framework, the approximate size would be 365 KB (uncompressed).
The engine itself take 86 KB in size.

If all of them got compressed, it would take around 104 KB.

**Reply by linonetwo** [reply]
Thank you for the answer and hard work!
